nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/CLICommand.java

```diff
@@ -911,33 +912,45 @@ protected void prevalidate() throws CommandException {
         /*
          * Check for missing options and operands.
          */
-        Console cons = programOpts.isInteractive() ? System.console() : null;
+        ConsoleReader cons = null;
+        if (programOpts.isInteractive()) {
+            try {
+                cons = new ConsoleReader(System.in, System.out, null);
+            } catch (IOException ioe) {
+                logger.log(Level.WARNING, "Error instantiating console", ioe);
+            }
+        }
 
         boolean missingOption = false;
         for (ParamModel opt : commandModel.getParameters()) {
             if (opt.getParam().password())
                 continue;       // passwords are handled later
-	    if (opt.getParam().obsolete() && getOption(opt.getName()) != null)
-		logger.info(strings.get("ObsoleteOption", opt.getName()));
-            if (opt.getParam().optional())
-                continue;
-            if (opt.getParam().primary())
-                continue;
-            // if option isn't set, prompt for it (if interactive)
-            if (getOption(opt.getName()) == null && cons != null && !missingOption) {
-                cons.printf("%s", strings.get("optionPrompt", lc(opt.getName())));
-                String val = cons.readLine();
-                if (ok(val)){
-                    options.set(opt.getName(), val);
+            if (opt.getParam().obsolete() && getOption(opt.getName()) != null) {
+                logger.info(strings.get("ObsoleteOption", opt.getName()));
+                if (opt.getParam().optional())
```

Comment:
Looks like this should be outside of `if (opt.getParam().obsolete() && getOption(opt.getName()) != null) {`
perhaps the bad formatting in the file before confused things?

Reply:
because of the addition of a {